src/lib/concurrency/transaction_context.hpp

```diff
@@ -53,7 +54,7 @@ class TransactionContext : public std::enable_shared_from_this<TransactionContex
   friend class TransactionManager;
 
  public:
-  TransactionContext(TransactionID transaction_id, CommitID snapshot_commit_id);
+  TransactionContext(TransactionID transaction_id, CommitID snapshot_commit_id, bool is_auto_commit = true);
```

Comment:
I don't think so. One argument could be: Auto-commit Transactions are created without a specific user command. Now these transactions don't explicitly need to state their implicity in the constructor, since auto-commit is the standard behavior.